test/helpers/browser.js

```diff
@@ -7,6 +7,7 @@ var vm = require('vm');
 
 function createContext() {
   var context = {
+    self: this,
```

Comment:
This is almost certainly wrong. If `self` is expected to be set to `window` (because that's the global `this` in browser), then you need to set `context.self = context`.

However, I would like to better understand what code is accessing the `this` variable? Can you paste source-mapped stack traces?